nemo/collections/tts/models/magpietts_preference_optimization.py

```diff
@@ -26,18 +26,20 @@
 from nemo.collections.tts.models import MagpieTTSModel
 
 
-class MagpieTTSModelPrefDataGen(MagpieTTSModel):
-    """Small override to save inference metrics, used for datagen in Offline PO"""
+class MagpieTTSModelInference(MagpieTTSModel):
```

Comment:
Your new class does not have the transcription option with whisper. Is this intentional? Compared to the older class in nemo/collections/tts/models/magpietts.py

Reply:
Yes. `transcribe_with_whisper` is now a stand alone function that is used both in the inference class and also GRPO class. So I have removed it from this class.